### PR TITLE
Disable cert rotation for temporary kube client

### PR DESCRIPTION
**Pull request by elgnay**
It's related to https://github.com/open-cluster-management/backlog/issues/5475.

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go [hunk 1]
```diff
@@ -303,6 +303,19 @@ func buildKubeClientWithSecret(secret *corev1.Secret) (kubernetes.Interface, err
 	if err != nil {
 		return nil, err
 	}
+
+	// use embeded cert/key data instead of references to external cert/key files to
```
Comment (qiujian16) [inline pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go]: do we still need line  291-304? if we read from secret directly, i think you could  build restConfig using  byte  directly by  using clientcmd.Load()

Comment (elgnay) [reply pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go]: Code refactored as suggested.

**Pull request merged**